compiler/fir/tree/src/org/jetbrains/kotlin/fir/expressions/impl/FirResolvedArgumentList.kt

```diff
@@ -21,4 +24,9 @@ class FirResolvedArgumentList internal constructor(
             argument.accept(visitor, data)
         }
     }
+
+    override fun <D> transformArguments(transformer: FirTransformer<D>, data: D): FirArgumentList {
+        arguments.forEach { it.transformSingle(transformer, data) }
```

Comment:
This solution doesn't satisfies contract of transformations. If transformer returns new instance of argument then it won't be written in `mapping`. Please replace it with this:
```
mapping = mapping.mapKeys { (it, _) -> it.transformSingle(transformer, data) }
```

And don't forget to make `mapping` mutable with private setter

Reply:
Also, good to learn these (and contract behind transformations). Edited as per suggestions.